includes/class-wc-swedbank-plugin.php

```diff
@@ -0,0 +1,554 @@
+	 */
+	public static $background_process;
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		// Includes
+		$this->includes();
+
+		// Activation
+		register_activation_hook( __FILE__, array( $this, 'install' ) );
+
+		// Actions
+		add_filter( 'plugin_action_links_' . plugin_basename( __FILE__ ), array( $this, 'plugin_action_links' ) );
+		add_action( 'plugins_loaded', array( $this, 'init' ), 0 );
+		add_action( 'woocommerce_init', array( $this, 'woocommerce_init' ) );
+		add_action( 'woocommerce_loaded', array( $this, 'woocommerce_hook_loaded' ) );
+
+		// Add statuses for payment complete
+		add_filter(
+			'woocommerce_valid_order_statuses_for_payment_complete',
+			array( $this, 'add_valid_order_statuses' ),
+			10,
+			2
+		);
+
+		// Status Change Actions
+		//add_action( 'woocommerce_order_status_changed', __CLASS__ . '::order_status_changed', 10, 4 );
+
+		// Add meta boxes
+		add_action( 'add_meta_boxes', __CLASS__ . '::add_meta_boxes' );
+
+		// Add action buttons
+		add_action( 'woocommerce_order_item_add_action_buttons', __CLASS__ . '::add_action_buttons', 10, 1 );
+
+		// Add scripts and styles for admin
+		add_action( 'admin_enqueue_scripts', __CLASS__ . '::admin_enqueue_scripts' );
+
+		// Add Admin Backend Actions
+		add_action( 'wp_ajax_swedbank_pay_capture', array( $this, 'ajax_swedbank_pay_capture' ) );
+
+		add_action( 'wp_ajax_swedbank_pay_cancel', array( $this, 'ajax_swedbank_pay_cancel' ) );
+
+		// Filters
+		add_filter( 'swedbank_pay_generate_uuid', array( $this, 'generate_uuid' ), 10, 1 );
+		add_filter( 'swedbank_pay_payment_description', array( $this, 'payment_description' ), 10, 2 );
+
+		// Process swedbank queue
+		if ( ! is_multisite() ) {
+			add_action( 'customize_save_after', array( $this, 'maybe_process_queue' ) );
+			add_action( 'after_switch_theme', array( $this, 'maybe_process_queue' ) );
+		}
+
+		// Add admin menu
+		add_action( 'admin_menu', array( $this, 'admin_menu' ), 99 );
+
+		// Add Upgrade Notice
+		if ( version_compare( get_option( 'woocommerce_payex_psp_version', '1.2.0' ), '1.2.0', '<' ) ) {
+			add_action( 'admin_notices', __CLASS__ . '::upgrade_notice' );
+		}
+	}
+
+	public function includes() {
+		$vendors_dir = dirname( __FILE__ ) . '/../vendors';
+
+		//if ( ! class_exists( '\\SwedbankPay\\Api\\Client\\Client', false ) ) {
+		//require_once $vendors_dir.'/swedbank-pay-sdk-php/vendor/autoload.php';
+		//}
+
+		spl_autoload_register(
+			function ( $class_name ) {
+				$auto_load_path = dirname( __FILE__ ) . '/../vendors/swedbank-pay-sdk-php/src/';
+				$class_file     = $auto_load_path . str_replace( '\\', DIRECTORY_SEPARATOR, $class_name ) . '.php';
+				if ( file_exists( $class_file ) ) {
+					  require_once $class_file;
+
+					  return true;
+				}
+
+				return false;
+			}
+		);
+
+		spl_autoload_register(
+			function ( $class_name ) {
+				$auto_load_path = dirname( __FILE__ ) . '/../vendors/swedbank-pay-core-library/src/';
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function dirname() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)